src/backend/marsha/core/api.py

```diff
@@ -175,3 +188,66 @@ def upload_policy(self, request, pk=None):
         )
 
         return Response(policy)
+
+
+class XAPIStatementView(APIView):
+    """Viewset managing xAPI requests."""
+
+    permission_classes = [IsVideoToken]
+    http_method_names = ["post"]
+
+    def post(self, request):
+        """Send a xAPI statement to a defined LRS.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint.
+            It contains a JSON representing part of the xAPI statement
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse to reflect if the XAPI request failed or is successful
+
+        """
+        if settings.LRS_URL is None:
+            return Response(
+                {"reason": "LRS is not configured. This endpoint is not usable."},
+                status=501,
+            )
+
+        xapi_statement = XAPIStatementSerializer(data=request.data)
+
+        if not xapi_statement.is_valid():
+            return Response(xapi_statement.errors, status=400)
+
+        lti_user = LTIUser(request.user)
+        try:
+            video = Video.objects.get(pk=lti_user.video_id)
+        except Video.DoesNotExist:
+            return Response(
+                {
+                    "reason": "video with id {id} does not exists".format(
+                        id=lti_user.video_id
+                    )
+                },
+                status=404,
+            )
+
+        xapi = XAPI(
+            settings.LRS_URL, settings.LRS_AUTH_TOKEN, settings.LRS_XAPI_VERSION
+        )
+
+        try:
+            xapi.send(video, xapi_statement.validated_data, lti_user)
+        # pylint: disable=invalid-name
+        except requests.exceptions.HTTPError as e:
+            message = "Impossible to send xAPI request to LRS."
+            logger.critical(
+                message,
+                extra={"response": e.response.text, "status": e.response.status_code},
+            )
+            return Response({"status": message}, status=400)
```

Comment:
should it be a 501 since it is for server reasons not the format of the data sent by the client?

Reply:
Yes you are right